modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/http4s/JwtAuthentication.scala

```diff
@@ -1,23 +1,63 @@
 package edu.gemini.seqexec.web.server.http4s
 
+import java.time.Instant
+
 import edu.gemini.seqexec.model.UserDetails
 import edu.gemini.seqexec.web.server.security.AuthenticationService
-import org.http4s.AttributeKey
+import org.http4s.{AttributeKey, Cookie, HttpService, Response}
 
 import scalaz._
 import Scalaz._
 import scalaz.concurrent.Task
 
+
+trait CookiesService {
+  def name: String
+  def ssl: Boolean
+  def ttl: Long
+
+  def buildCookie(token: String): Cookie = {
+    // if successful set a cookie
+    val exp = Instant.now().plusSeconds(ttl)
```

Comment:
This does unsafe IO. I think in this particular case there is no need to correct it because it seems `buildCookie` is only meant to be used locally and the Tasks where it's being called are executed right away or close to it, but it could be a problem if you use it to build more complex Task trees.

It seems that in Scala code by convention time measurements are unsafe, but I'd just would wrap every time measurement in a Task, just to avoid thinking about potential issues down the line.